contracts/truefi2/DebtToken.sol

```diff
@@ -4,37 +4,108 @@ pragma solidity 0.6.10;
+    uint256 public override debt;
+
+    address public liquidator;
+    ITrueFiPool2 public override pool;
+    ERC20 public override token;
+
+    uint256 public redeemed;
+
+    Status public override status;
+
+    /**
+     * @dev Emitted when a DebtToken is redeemed for underlying tokens
+     * @param receiver Receiver of tokens
+     * @param burnedAmount Amount of DebtTokens burned
+     * @param redeemedAmount Amount of token received
+     */
+    event Redeemed(address receiver, uint256 burnedAmount, uint256 redeemedAmount);
+
+    /**
+     * @dev Emitted when loan gets liquidated
+     * @param status Final loan status
+     */
+    event Liquidated(Status status);
+
+    function initialize(
+        ITrueFiPool2 _pool,
+        address _lender,
+        address _borrower,
+        address _liquidator,
+        uint256 _debt
+    ) external initializer {
+        ERC20.__ERC20_initialize("TrueFi Debt Token", "DEBT");
+
+        pool = _pool;
+        token = _pool.token();
+        borrower = _borrower;
+        liquidator = _liquidator;
+        debt = _debt;
+        status = Status.Defaulted;
+        _mint(_lender, _debt);
+    }
+
+    /**
+     * @dev Redeem DebtToken balances for underlying token
+     * Can only call this function after the loan is Closed
+     * @param _amount amount to redeem
+     */
+    function redeem(uint256 _amount) external override {
+        uint256 amountToReturn = _amount.mul(_balance()).div(totalSupply());
+        redeemed = redeemed.add(amountToReturn);
+        _burn(msg.sender, _amount);
+        token.safeTransfer(msg.sender, amountToReturn);
+
+        emit Redeemed(msg.sender, _amount, amountToReturn);
+    }
+
+    /**
+     * @dev Liquidate the loan if it has defaulted
+     */
+    function liquidate() external override {
+        require(status == Status.Defaulted, "DebtToken: Current status should be Defaulted");
+        require(msg.sender == liquidator, "DebtToken: Caller is not the liquidator");
+
+        status = Status.Liquidated;
+
+        emit Liquidated(status);
+    }
+
+    /**
+     * @dev Check how much was already repaid
+     * Funds stored on the contract's address plus funds already redeemed by lenders
+     * @return Uint256 representing what value was already repaid
+     */
+    function repaid() external override view returns (uint256) {
+        return _balance().add(redeemed);
+    }
+
+    /**
+     * @dev Public currency token balance function
+     * @return token balance of this contract
+     */
+    function balance() external override view returns (uint256) {
+        return _balance();
+    }
+
+    function version() external override pure returns (uint8) {
+        return 1;
+    }
+
+    /**
+     * @dev Get currency token balance for this contract
+     * @return token balance of this contract
+     */
+    function _balance() internal view returns (uint256) {
+        return token.balanceOf(address(this));
+    }
 }
```

Comment:
Agreed. It's unclear yet exactly what mechanism litigation will use to force debt repayment (fiat to banks vs. ERC20 transfer to this LoanToken vs. ERC20 transfer to another contract -- it depends on jurisdiction + terms of settlement), so I'm inclined to avoid adding these yet.

+1 for creating tasks so we can bring these up for discussion though.

(My opinion) I'm inclined to avoid reintroducing `reclaim()` at all:
1. If a borrower accidentally overpays then we can `redeem()` and manually return funds to them.
2. During litigation, legal counsel would likely demand interest be accrued on top of the original debt amount owed. If the borrower ends up settling with an ERC20 transfer to the DebtToken, we wouldn't want them to be able to withdraw that interest via `reclaim()`.